langchain/src/agents/structured_chat/outputParser.ts

```diff
@@ -0,0 +1,72 @@
+import { AgentActionOutputParser } from "../types.js";
+import { AGENT_ACTION_FORMAT_INSTRUCTIONS, FORMAT_INSTRUCTIONS } from "./prompt.js";
+import { OutputFixingParser } from "../../output_parsers/fix.js";
+import { BaseLanguageModel } from "../../base_language/index.js";
+import { AgentAction, AgentFinish } from "../../schema/index.js";
+import { OutputParserException } from "../../schema/output_parser.js";
+
+export class StructuredChatOutputParser extends AgentActionOutputParser {
+  async parse(text: string): Promise<AgentAction | AgentFinish> {
+    try {
+      const regex = /```(?:json)?(.*)(```)/gs;
+      const actionMatch = regex.exec(text);
+      if (actionMatch === null) {
+        throw new OutputParserException(`Could not parse an action. The agent action must be within a markdown code block, and "action" must be a provided tool or "Final Answer"`);
+      }
+      const response = JSON.parse(actionMatch[1].trim());
+      const { action, action_input } = response;
+
+      if (action === "Final Answer") {
+        return { returnValues: { output: action_input }, log: text };
+      }
+      return { tool: action, toolInput: action_input || {}, log: text };
+    } catch (e) {
+      throw new OutputParserException(
+        `Failed to parse. Text: "${text}". Error: ${e}`
+      );
+    }
+  }
+
+  getFormatInstructions(): string {
+    return AGENT_ACTION_FORMAT_INSTRUCTIONS;
```

Comment:
This is incorrect, it's returning an unformatted template string, without replacing the template vars. Either remove template vars, or format the template